examples/gno.land/r/demo/pagination/pagination.gno

```diff
@@ -0,0 +1,137 @@
+// This realm demonstrates a pagination system working with gnoweb
+// It keeps a slice of signed up addresses for efficient pagination
+package pagination
+
+import (
+	"std"
+	"strconv"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/mux"
+	"gno.land/p/demo/ufmt"
+)
+
+type Signup struct {
+	account string
+	height  int64
+}
+
+// signups - keep a slice of signed up addresses efficient pagination
+var signups []Signup
+
+// tracker - keep track of who signed up
+var tracker *avl.Tree
+
+const pageSize = 20
+
+func init() {
+	tracker = avl.NewTree()
+	SignUp()
+}
+
+func SignUp() string {
+	caller := std.GetOrigCaller().String()
+	height := std.GetHeight()
+
+	_, exists := tracker.Get(caller)
+	if exists {
+		panic(caller + " is already signed up!")
+	}
+
+	tracker.Set(caller, struct{}{})
+	signup := Signup{
+		caller,
+		height,
+	}
+
+	signups = append(signups, signup)
+	return ufmt.Sprintf("%s signed up at block #%d!", signup.account, signup.height)
+}
+
+func GetSignedUpInRange(page, pageSize int) []Signup {
+	if page < 1 {
+		panic("page number cannot be less than 1")
+	}
+
+	if pageSize < 1 || pageSize >= 50 {
+		panic("page size must be between 1 and 50")
+	}
+
+	// Pagination
+	// Calculate indexes
+	startIndex := (page - 1) * pageSize
+	endIndex := startIndex + pageSize
+
+	// If page does not contain any posts
+	if startIndex >= len(signups) {
+		return nil
+	}
+
+	// If page contains fewer posts than the page size
+	if endIndex > len(signups) {
+		endIndex = len(signups)
+	}
+
+	return signups[startIndex:endIndex]
+}
+
+func renderHelper(res *mux.ResponseWriter, req *mux.Request) {
+	totalSignups := len(signups)
+
+	res.Write("# Welcome to Gno.land!\n\n")
+	res.Write(ufmt.Sprintf("## Signed up users (%d):\n\n", totalSignups))
+
+	page, err := strconv.Atoi(req.GetVar("number"))
```

Comment:
Might be better to make `"number"` a constant to be used both here and in `Render`.

Reply:
[a41d2bc](https://github.com/gnolang/gno/pull/1949/commits/a41d2bcc72c2aa0fa4cf3bad7216c7bccca796ff)